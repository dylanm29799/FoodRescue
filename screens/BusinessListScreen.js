import React from "react";
import {
  View,
  Text,
  StyleSheet,
  Button,
  FlatList,
  TouchableOpacity,
  Image,
  SafeAreaView,
} from "react-native";
import {
  scale,
  verticalScale,
  moderateScale,
} from "../components/ResponsiveText";
import Colour from "../constants/Colour";
import MapView from "react-native-maps";
import { BUSINESSLIST } from "../Data/businesslistDataExample";
import { FOODCOUNTDOWN } from "../Data/foodcountdowndataexample";
import { ScrollView } from "react-native-gesture-handler";

const BusinessListScreen = (props) => {
  const businessName = props.navigation.getParam("BusinessName");

  const renderCategory = (itemData) => {
    return (
      <TouchableOpacity
        style={styles.Categories}
        onPress={() => {
          props.navigation.navigate({
            routeName: "ItemDetail",
            params: {
              BusinessName: businessName,
              Title: itemData.item.title,
              Image: itemData.item.image,
              Quantity: itemData.item.quantity,
              InitialPrice: itemData.item.initialPrice,
              Price: itemData.item.price,
            },
          });
        }}
      >
        <View style={styles.item}>
          <Image
            style={{ height: "75%", width: "100%" }}
            source={{ uri: itemData.item.image }}
          />
          <View>
            <View style={styles.heading}>
              <Text style={styles.text1}>{itemData.item.title}</Text>
              <Text style={styles.text2}>
                {itemData.item.quantity} <Text>Items left!</Text>
              </Text>
            </View>

            <Text style={{ paddingLeft: "5%" }}>
              <Text style={styles.text3}>€{itemData.item.initialPrice}</Text>
              <Text style={styles.text4}>€{itemData.item.price}</Text>
            </Text>
          </View>
        </View>
      </TouchableOpacity>
    );
  };
  const renderFoodCategory = (itemData) => {
    return (
      <TouchableOpacity
        style={styles.Categories}
        onPress={() => {
          props.navigation.navigate({
            routeName: "ItemDetail",
            params: {
              BusinessName: businessName,
              Title: itemData.item.title,
              Image: itemData.item.image,
              Quantity: itemData.item.quantity,
              InitialPrice: itemData.item.initialPrice,
              Price: itemData.item.price,
              Time: itemData.item.time,
            },
          });
        }}
      >
        <View style={styles.item}>
          <Image
            style={{ height: "75%", width: "100%" }}
            source={{ uri: itemData.item.image }}
          />
          <View>
            <View style={styles.heading}>
              <Text style={styles.text1}>{itemData.item.title}</Text>

              <Text style={styles.text2}>
                {itemData.item.quantity} <Text>Items left!</Text>
              </Text>
            </View>
            <View style={{ flex: 1, flexDirection: "row" }}>
              <View
                style={{
                  paddingLeft: "5%",
                  flex: 2,
                }}
              >
                <Text style={styles.text4}>
                  €{itemData.item.price}
                  <Text style={styles.text3}>
                    €{itemData.item.initialPrice}
                  </Text>
                </Text>
              </View>
              <View style={{ flex: 1 }}>
                <Text
                  style={{
                    paddingRight: "9%",
                    fontSize: scale(15),
                    textAlign: "right",
                  }}
                >
                  {itemData.item.time} Hours Left!
                </Text>
              </View>
            </View>
          </View>
        </View>
      </TouchableOpacity>
    );
  };
  return (
    <View style={styles.screen}>
      <ScrollView>
        <View style={styles.MapBorder}>
          <MapView
            style={styles.Map}
            initialRegion={{
              latitude: 53.34900146651947,
              longitude: -6.243367195129395,
              latitudeDelta: 0,
              longitudeDelta: 0.05,
            }}
          />
        </View>

        <SafeAreaView style={{ alignItems: "center", flex: 1 }}>
          <Text style={styles.sortName}>Available Items</Text>

          <FlatList
            style={{ width: "90%", flex: 1 }}
            data={BUSINESSLIST}
            renderItem={renderCategory}
          />
        </SafeAreaView>

        <SafeAreaView
          style={{
            alignItems: "center",
            flex: 1,
            justifyContent: "space-between",
          }}
        >
          <Text style={styles.sortName}>Food Countdown</Text>
          <FlatList
            style={{ width: "90%", flex: 1 }}
            data={FOODCOUNTDOWN}
            renderItem={renderFoodCategory}
          />
        </SafeAreaView>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  screen: {
    flex: 1,
  },

  Map: {
    height: "100%",
    width: "100%",
  },

  MapBorder: {
    borderWidth: scale(2),
    borderColor: Colour.primaryColour,
    position: "relative",
    height: scale(200),
  },
  item: {
    borderWidth: scale(2),
    borderColor: Colour.primaryColour,
    height: scale(200),
  },
  heading: {
    flexDirection: "row",
  },
  text1: {
    width: "50%",
    paddingLeft: "5%",
    fontSize: scale(15),
  },
  text2: {
    width: "50%",
    textAlign: "right",
    paddingRight: "5%",
    fontSize: scale(15),
  },

  text3: {
    paddingLeft: "5%",
    fontSize: scale(15),
    textDecorationLine: "line-through",
    textDecorationStyle: "solid",
  },

  text4: {
    fontSize: scale(15),
  },

  sortName: {
    fontSize: scale(30),
    textAlign: "center",
    paddingVertical: scale(10),
    fontFamily: "OpenSans",
    fontWeight: "900",
  },
  Categories: { flex: 1 },
  row: {
    flex: 1,
    justifyContent: "space-around",
  },
});

export default BusinessListScreen;
