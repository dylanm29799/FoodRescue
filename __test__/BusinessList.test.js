// __tests__/BusinessList.test.js

import React from "react";
import BusinessListScreen from "../screens/BusinessListScreen";
import renderer from "react-test-renderer";
import { firebaseConfig } from "../config";
import * as firebase from "firebase";

firebase.initializeApp(firebaseConfig);
test("renders correctly", () => {
  const tree = renderer.create(<BusinessListScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
