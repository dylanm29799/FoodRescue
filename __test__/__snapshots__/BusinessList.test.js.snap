// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderColor": "#c44536",
        "borderWidth": 4.285714285714286,
        "height": 471.4285714285714,
        "position": "relative",
      }
    }
  >
    <AIRMap
      handlePanDrag={false}
      initialRegion={
        Object {
          "latitude": "",
          "latitudeDelta": 0.1,
          "longitude": "",
          "longitudeDelta": 0.1,
        }
      }
      onChange={[Function]}
      onLayout={[Function]}
      onMapReady={[Function]}
      onMarkerPress={[Function]}
      region={null}
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <AIRMapMarker
        coordinate={
          Object {
            "latitude": undefined,
            "longitude": undefined,
          }
        }
        description="Your Location"
        onPress={[Function]}
        stopPropagation={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "position": "absolute",
            },
            undefined,
          ]
        }
        title="You"
      >
        <Text />
      </AIRMapMarker>
      <AIRMapMarker
        coordinate={
          Object {
            "latitude": "",
            "longitude": "",
          }
        }
        onPress={[Function]}
        stopPropagation={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "position": "absolute",
            },
            undefined,
          ]
        }
        title=""
      >
        <Text />
      </AIRMapMarker>
    </AIRMap>
  </View>
  <RCTScrollView
    style={
      Object {
        "backgroundColor": "#ffecd2",
      }
    }
  >
    <View>
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4294962386,
              0,
            ],
            "endPoint": undefined,
            "locations": undefined,
            "startPoint": undefined,
          }
        }
        style={
          Object {
            "borderRadius": 15,
          }
        }
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "borderBottomWidth": 1,
                "borderTopWidth": 1,
                "color": "#444444",
                "fontFamily": "MonM",
                "fontSize": 42.857142857142854,
                "paddingVertical": 10.714285714285714,
                "textAlign": "center",
                "width": "100%",
              }
            }
          >
            Products To Buy
          </Text>
          <RCTScrollView
            data={Object {}}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={true}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            style={
              Object {
                "flex": 1,
              }
            }
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <View />
          </RCTScrollView>
        </RCTSafeAreaView>
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "borderBottomWidth": 1,
                "borderTopWidth": 1,
                "color": "#444444",
                "fontFamily": "MonM",
                "fontSize": 42.857142857142854,
                "paddingVertical": 10.714285714285714,
                "textAlign": "center",
                "width": "100%",
              }
            }
          >
            Food Countdown
          </Text>
          <RCTScrollView
            data={Object {}}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={true}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            style={
              Object {
                "flex": 1,
                "paddingBottom": 10,
              }
            }
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <View />
          </RCTScrollView>
        </RCTSafeAreaView>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </RCTScrollView>
</View>
`;
